Tango ROS nodelet to node

mcopejans opened the pull request.

PerrineAguiar commented on TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/TangoServiceClientNode.java:
```diff
@@ -84,80 +91,117 @@ public void onNewMessage(Int8 status) {
+
+    public Boolean getMapUuids() throws ServiceNotFoundException {
+        ServiceClient<GetMapUuidsRequest, GetMapUuidsResponse> saveMapService =
+                mConnectedNode.newServiceClient(SAVE_MAP_SRV_NAME, SaveMap._TYPE);
```
nit: GET_MAP_UUIDS_SRV_NAME

PerrineAguiar commented on TangoRosStreamer/nodelet_manager/src/main/java/eu/intermodalics/nodelet_manager/TangoInitializationHelper.java:
```diff
@@ -122,70 +122,70 @@ public static final int loadTangoSharedLibrary() {
             basePath = "/data/data/com.projecttango.tango/libfiles/";
         }
-        Log.i("TangoInitializationHelper", "basePath: " + basePath);
+        Log.i("TangoInitializationHelp", "basePath: " + basePath);
```
Why not keeping TangoInitializationHelper?

mcopejans replied:
Android studio complained on this. The tag can have maximum 23 characters. Would you prefer a different tag? E.g. just TangoInitialization?

PerrineAguiar replied:
Ok, it is fine like this, no need to change. I was just wondering.

PerrineAguiar commented on tango_ros_common/tango_ros_native/include/tango_ros_native/tango_ros_node.h:
```diff
@@ -151,9 +156,14 @@ class TangoRosNode {
   // Function called when one of the dynamic reconfigure parameter is changed.
   // Updates the publisher configuration consequently.
   void DynamicReconfigureCallback(PublisherConfig &config, uint32_t level);
+  bool GetMapName(const tango_ros_messages::GetMapName::Request& req,
```
Add doc?

PerrineAguiar commented on tango_ros_common/tango_ros_native/include/tango_ros_native/tango_ros_node.h:
```diff
@@ -151,9 +156,14 @@ class TangoRosNode {
   void DynamicReconfigureCallback(PublisherConfig &config, uint32_t level);
+  bool GetMapName(const tango_ros_messages::GetMapName::Request& req,
+                  tango_ros_messages::GetMapName::Response &res);
+  bool GetMapUuids(const tango_ros_messages::GetMapUuids::Request &req,
```
Add doc?

PerrineAguiar commented on TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/TangoServiceClientNode.java:
```diff
@@ -84,80 +91,117 @@ public void onNewMessage(Int8 status) {
     }
 
-    public boolean callSaveMapService(String mapName) {
+    public boolean wrapCallROSService(String serviceName, Callable<Boolean> callable) {
```
Can we make this private?

PerrineAguiar commented on TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/TangoServiceClientNode.java:
```diff
@@ -84,80 +91,117 @@ public void onNewMessage(Int8 status) {
-            System.out.println("Client node not ready: " + TANGO_CONNECT_SRV_NAME);
-            return false;
-        }
+    public Boolean tangoConnect(final byte connectRequest) throws ServiceNotFoundException {
```
Can we make this private?

PerrineAguiar commented on TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/TangoServiceClientNode.java:
```diff
@@ -84,80 +91,117 @@ public void onNewMessage(Int8 status) {
 
-    private String BuildTangoRosNodeNamespaceName(String paramName) {
-        return "/" + TangoRosNode.NODE_NAME + "/" + paramName;
+    public Boolean saveMap(String mapName) throws ServiceNotFoundException {
```
Same here, can we make this private?

PerrineAguiar commented on TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/TangoServiceClientNode.java:
```diff
@@ -84,80 +91,117 @@ public void onNewMessage(Int8 status) {
+        return true;
+    }
+
+    public Boolean getMapUuids() throws ServiceNotFoundException {
```
Can we make this private?

mcopejans commented:
@PerrineAguiar @smits Thanks for your reviews. I addressed all comments. I'm not going to merge this into the base branch since this will get a huge PR otherwise. Feel free to merge whenever it suits you.
